src/Microsoft.PowerShell.Commands.Utility/commands/utility/CSVCommands.cs

```diff
@@ -1577,10 +1572,7 @@ private string
                 else if (IsNewLine(ch))
                 {
                     endOfRecord = true;
-                    if (ch == '\r')
-                    {
-                        ReadChar();
-                    }
+                    SkipNewLine();
```

Comment:
If there are two newlines, won't this skip the second?

A better pattern might be to change `IsNewLine` to return a string (`"\n"`, `"\r"`, or `"\r\n"`) or null, and if the newline must be appended, append the result. There should be no need for `SkipNewLine`.

Reply:
I don't know about skipping empty lines. Formally [RFC4180](https://www.ietf.org/rfc/rfc4180.txt) don't allow empty lines. Although practically it may make sense.

I thought about the pattern ( `IsNewLine`  to return a string ) and decided to begin with `SkipNewLine()` as seems more readable.  Now I'm going to implement `IsNewLine(out newLine)`.

Reply:
I test with Windows PowerShell - we keep its behaviour and ignore empty lines.

Microsoft Excel imports empty lines as empty rows. We should discuss this in new Issue if we want because it is a breaking change.